### ISSUE TITLE
wiki: Linode

### ISSUE THREAD

**Issue by garbas** [NixOS/nixpkgs#13177]
https://nixos.org/wiki/Linode

**Comment by nixy** [comment]
I would be willing to rewrite the documentation for this page as it is quite old. It also sets up NixOS in a way that doesn't play well with Linode's tools. I think a rewrite of this is necessary and would be helpful, but I am not sure where the new documentation would go.

Would this kind of documentation be appropriate for the NixOS manual or perhaps the downloads page? Is there some sort of "Things you can run NixOS on" page?

**Comment by Mic92** [comment]
@nixy yes this should be definitely in the NixOS manual.

**Comment by nixy** [comment]
I have a draft done and I am working on making some improvements to the NixOS experience on Linode as a whole.

In particular all of Linode's tools don't support partitioned disks, as they exclusively use partitionless disks. NixOS currently doesn't support this, but #19379 once merged will allow NixOS to be installed on a partitionless disk. This will let all of Linode's tools such as backups and password resets work with NixOS.

I am also considering making a script to automate the process and provide a "basic" configuration automatically. That would be a installation of NixOS with the normal Linode tools, that is sshd with root access allowed and some other goodies.

This should make using NixOS on Linode much more pleasant. But, this means the guide is a little meaty. I think it might be a bit too large for the NixOS manual directly. Is it possible to break the manual into pages and link to those pages? This will allow the documentation to go more in depth about particular areas and provide examples/walkthroughs without making the manual unreadable.

**Comment by garbas** [comment]
@nixy here is how we are moving with the manual. clearly not everything should be in there even at the current state, but this is currently the best way to keep all resources in one place. once wiki pages are migrated (hopefully soon) we should start thinking about reorganizing the manual(s) to make it also more appealing for newcomers. having all the documentation in one place will make this very easy.

so in short, put the instructions in manual and we'll refractor the manual at later stage.

**Comment by nixy** [comment]
So its been a while, but I am still working on this but I am just trying to get NixOS in a state that will run nicely on Linode. #19379 was was accepted into nixpkgs, but I would need it to be in a release in order to use this for a guide.

Using the current guide and partitioning your Linode's disk leaves you with a fairly crippled installation. You won't ever be able to resize your filesystem, reset your password, or use Linode's backup services.

Is there a "proper" way to get things merged from unstable into a release? Looking at the manual it seems like the only way to get something into a release is to wait for the next release to come around.

**Comment by Mic92** [comment]
As you already have to edit configuration.nix before installing,
you could also enter this snippet instead:

```
# /etc/nixos/configuration.nix
    packageOverrides = pkgs: rec {                                                                                                                                                             
      grub2 = pkgs.grub2.overrideDerivation (oldAttrs: {                                                                                                                                       
        buildInputs = oldAttrs.buildInputs ++ [pkgs.makeWrapper];                                                                                                                              
        preFixup = ''                                                                                                                                                                          
          wrapProgram "$out/sbin/grub-install" --add-flags "--force"                                                                                                                           
        '';                                                                                                                                                                                    
      }); 
    };
```

**Comment by nixy** [comment]
@Mic92 you are my hero. That works like a charm.

I am gonna go ahead and finish up the guide and make a PR for the NixOS manual. Thanks for all the help.

**Comment by Ekleog** [comment]
(triage) The wiki is now dead. @nixy, do you still plan on writing a guide for the NixOS manual?

**Comment by nixy** [comment]
I wrote [this guide](https://linode.com/docs/tools-reference/custom-kernels-distros/install-nixos-on-linode/) for Linode while I worked for them and am actively maintaining it with new releases.

I feel like linking to this from the NixOS manual would be the neatest way to handle this while advertising it as an available platform. I'll take a look at adding it to the manual sometime this weekend.
